### ISSUE TITLE
[Apex] ApexDoc rule should not require "@description"

### ISSUE THREAD

**Issue by SCWells72** [pmd/pmd#3566]
**Affects PMD Version:** 6.39.0

**Rule:** ApexDoc

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/pmd-6.39.0/pmd_rules_apex_documentation.html#apexdoc

**Description:**

While standard ApexDoc does require the `@description` tag to denote the beginning of description text, the [`SfApexDoc` variant](https://gitlab.com/StevenWCox/sfapexdoc) enhances ApexDoc in a number of ways including removing the need for the `@description` tag. This makes ApexDoc more consistent with JavaDoc, ESDoc, etc., where all text in the doc comment before the first tag is considered description text.

An option should be added to the `ApexDoc` rule to disable the `@description` tag requirement for ApexDoc comments.

**Code Sample demonstrating the issue:**

```apex
/**
 * This is all considered description text by SfApexDoc just as it is in JavaDoc and similar even 
 * without the need for an explicit tag.
 * 
 * @see something else
 */
```

**Expected outcome:**

If the `ApexDoc` rule is configured with defaults, an issue is reported for the documentation comment above that `@description` is missing. If it is configured as follows (using `reportMissingDescriptionTag` as a suggestion), no such issue is reported:

```xml
   <rule ref="category/apex/documentation.xml/ApexDoc" message="Document classes, methods, and properties that are public or global.">
      <priority>3</priority>
      <properties>
         <property name="reportMissingDescriptionTag" value="false" />
      </properties>
   </rule>
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
CLI and direct API-based integration
